### PR TITLE
Notify apm of exceptions

### PR DESCRIPTION
**Pull request by Widdershin**
This branch adds integration with `raygun-apm` that allows for linked exceptions in the Raygun app.

We now generate a UUIDv4 to serve as a correlation id, as pass that both to the crash reporting endpoint and to the apm module. This allows the generated trace to reference the linked exception correctly.

### CODE REVIEW

File: lib/raygun.ts [hunk 1]
```diff
@@ -30,8 +30,23 @@ import { OfflineStorage } from "./raygun.offline";
...
+try {
+  if (module.parent) {
+    apmBridge = module.parent.require('raygun-apm/lib/src/crash_reporting');
+  } else {
+    apmBridge = require('raygun-apm/lib/src/crash_reporting');
+  }
+} catch (e) {
+  apmBridge = null;
+}
+
```
Comment (Widdershin) [inline lib/raygun.ts:49]: This looks a little complex, but in essence we need require the `raygun-apm` module from the parent's require scope to have access. If for any reason we aren't able to do that, linked exceptions will be disabled.

File: lib/raygun.ts [hunk 2]
```diff
@@ -248,6 +269,8 @@ class Raygun {
           : message;
     }
 
+    message.details.correlationId = correlationId;
```
Comment (BenjaminHarding) [inline lib/raygun.ts:272]: Should the message always have a `correlationId` or only if the `apmBridge` has been used?

Comment (Widdershin) [reply lib/raygun.ts:272]: Good point, might as well save the bytes if the `apmBridge` isn't present :)

**Comment by Widdershin** [comment]
Thanks @BenjaminHarding for the snappy review 😸

**Comment by BenjaminHarding** [comment]
Anytime :D

**Pull request merged**